client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,388 @@
+
+const Nav = styled.div`
+  background: ${prop('MobilePanel.default.background')};
+  color: ${prop('primaryTextColor')};
+  padding: ${remSize(8)} 0;
+  gap: ${remSize(14)};
+  display: flex;
+  align-items: center;
+  font-size: ${remSize(10)};
+  box-shadow: #00000030 0px 2px 8px 0px;
+  z-index: 10;
+`;
+
+const LogoContainer = styled.div`
+  width: ${remSize(28)};
+  aspect-ratio: 1;
+  margin-left: ${remSize(14)};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  > svg {
+    width: 100%;
+    height: 100%;
+    > path {
+      fill: ${prop('accentColor')};
+      stroke: ${prop('accentColor')};
+    }
+  }
+`;
+
+const Title = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${remSize(2)};
+
+  > * {
+    padding: 0;
+    margin: 0;
+  }
+
+  > h5 {
+    font-size: ${remSize(13)};
+    font-weight: normal;
+  }
+`;
+
+const Options = styled.div`
+  margin-left: auto;
+  display: flex;
+  /* transform: translateX(${remSize(12)}); */
+  svg {
+    fill: ${(props) => props.color};
+  }
+
+  /* Drop Down menu */
+
+  > div > button:focus + ul,
+  > div > ul > button:focus ~ div > ul {
+    transform: scale(1);
+    opacity: 1;
+  }
+
+  > div {
+    position: relative;
+    ul {
+      ${prop('MobilePanel.default')}
+      position: absolute;
+      overflow: hidden;
+      z-index: 10;
+      right: 10px;
+      transform: translateX(${remSize(6)});
+      width: max-content;
+      transform: scale(0);
+      opacity: 0;
+      transform-origin: top;
+      transition: opacity 100ms;
+      transition: transform 100ms 200ms;
+      display: flex;
+      flex-direction: column;
+      gap: ${remSize(2)};
+      box-shadow: rgba(60, 64, 67, 0.3) 0px 1px 2px 0px,
+        rgba(60, 64, 67, 0.15) 0px 1px 3px 1px;
+      min-width: ${remSize(120)};
+      border-radius: ${remSize(2)};
+
+      /* User */
+      li.user {
+        color: ${prop('colors.processingBlue')};
+        font-size: ${remSize(16)};
+        font-weight: bold;
+        margin: ${remSize(6)} ${remSize(12)};
+      }
+
+      > b {
+        margin: ${remSize(6)} ${remSize(12)};
+        position: relative;
+        width: min-content;
+        &::after {
+          content: '';
+          position: absolute;
+          opacity: calc(0.5);
+          top: 50%;
+          transform: translate(100%, -50%);
+          right: ${remSize(-6)};
+          width: ${remSize(55)};
+          height: 1px;
+          background-color: ${prop('Button.primary.default.foreground')};
+        }
+      }
+
+      > li {
+        display: flex;
+        width: 100%;
+        > button {
+          width: 100%;
+          padding: ${remSize(8)} ${remSize(15)} ${remSize(8)} ${remSize(10)};
+          font-size: ${remSize(18)};
+          text-align: left;
+          &:hover {
+            background-color: ${prop('Button.primary.hover.background')};
+            color: ${prop('Button.primary.hover.foreground')};
```

Comment:
If this style is very similar to the primary Button then it might possibly make sense to have these buttons be an extension of it. `styled(Button)`.  You can style components which are already styled, so long as they pass through a `className` prop.

Reply:
yeah, but it just has same hover colors, rest is totally different so Idk if extending really makes sense, let me know

Reply:
Ok in that case it does not make sense.

I'm still figuring out how to best use styled-components in a clean and composable way.  Like what's the equivalent of the `@mixin` structure in the current SCSS, so we could do things like apply a "button hover" mixin.

Reply:
yeah